main.go

```diff
@@ -187,10 +189,40 @@ func main() {
 		"Duration the Leader elector clients should wait between tries of actions.")
 	flag.Parse()
 
+	mgrNamespace := ""
+	trueString := "true"
+
+	// Controller can be in namespace scope mode depending on an env variable
+	namespaceScope := os.Getenv(NamespaceScopeEnvVar)
+	setupLog.Info("==============", "namespaceScope", namespaceScope)
+
+	// Here we check whether RBAC is set for cluster scope
+	err = cl.Get(context.Background(), client.ObjectKey{Name: "foo"}, &corev1.Namespace{})
+	clusterScopeEnabled = err == nil || errors.IsNotFound(err)
+
+	if clusterScopeEnabled {
+		if namespaceScope != trueString {
+			setupLog.Info("Controller operating in cluster scope mode, will attempt to watch all namespaces")
+		} else {
+			// Config mismatch, namespace mode with cluster permissions, exit
+			setupLog.Info("In namespace scope mode but controller has cluster scope permissions, exit")
+			os.Exit(1)
```

Comment:
I'm not sure we need to fail in this case, just have more permission than needed.

Reply:
This might cause unexpected behaviors. The user would expect the controller to watch and reconcile in a single namespace but the controller is acting across all namespaces. I think we should exit and make sure the user/installer syncs up RBAC with install option. What do you think?

Reply:
Additionally clusterScopeEnabled will affect how service and SR controllers work internally. For ex, clusterScopeEnabled = true and namespaceScope = true would have different code path from clusterScopeEnabled = false and namespaceScope = true

Reply:
@chinhuang007 OK sorry I think the thing I meant to say but forgot to add is that we should also change the later logic to depend on the namespaceScope setting instead of this RBAC check. Basically change the two places where `clusterScopeEnabled` is referenced lower down in this func. Are you ok with the above change if we also do that?